.github/workflows/_ci.yml

```diff
@@ -96,21 +96,26 @@ jobs:
       with:
         image_tag: ${{ inputs.cache_build_image }}
 
-    # If the run is "trusted" (ie has access to secrets) then it should
-    # **not** set the ref and should use the code of the calling context.
-    - if: ${{ inputs.repo_ref && inputs.trusted }}
-      run: |
-        echo '`repo_ref` should not be set for trusted CI runs'
-        exit 1
-
     - uses: actions/checkout@v4
       name: Checkout Envoy repository
       with:
-        fetch-depth: ${{ inputs.repo_fetch_depth }}
+        fetch-depth: ${{ ! inputs.trusted && inputs.repo_fetch_depth || 0 }}
         # WARNING: This allows untrusted code to run!!!
         #  If this is set, then anything before or after in the job should be regarded as
         #  compromised.
         ref: ${{ ! inputs.trusted && inputs.repo_ref || '' }}
+
```

Comment:
Not for this PR:
I wonder if it's possible to create a list somewhere of all jobs that are trusted, so there will be a single point that defines which ones are considered "trusted-only" and abort anything that is untrusted and attempts to run paths that are considered non-trusted.

Reply:
the risk with untrusted code being run in a trusted env is from the point where you check it out

to mittigate above i initially didnt allow code that was running in a trusted env to set its sha

this caused a load of issues so this is another, more directed, attempt to restrict that situation

for RBE (and actually cache in gh) reasons we need to update our policy and get this workflow right so it will defo iterate